### ISSUE TITLE
intellji IDEA: No decompiled source function with name (__info__)

### ISSUE THREAD

**Issue by depeiwang** [KronicDeth/intellij-elixir#1883]
# Version

11.9.1

# What I was doing

I don't know what I was doing

# Exceptions

### Stacktrace

```
java.lang.Throwable: No decompiled source function with name (__info__)
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:159)
	at org.elixir_lang.beam.psi.impl.ModuleImpl.setMirror(ModuleImpl.kt:68)
	at org.elixir_lang.beam.psi.impl.ModuleElementImpl.setMirror(ModuleElementImpl.java:55)
	at org.elixir_lang.beam.psi.impl.ModuleElementImpl.setMirrors(ModuleElementImpl.java:46)
	at org.elixir_lang.beam.psi.impl.ModuleElementImpl.setMirrors(ModuleElementImpl.java:37)
	at org.elixir_lang.beam.psi.BeamFileImpl.setMirror(BeamFileImpl.java:634)
	at org.elixir_lang.beam.psi.BeamFileImpl$1.run(BeamFileImpl.java:607)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:658)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:610)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:65)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeNonCancelableSection(CoreProgressManager.java:218)
	at org.elixir_lang.beam.psi.BeamFileImpl.getMirror(BeamFileImpl.java:605)
	at org.elixir_lang.beam.psi.impl.ModuleElementImpl.getMirror(ModuleElementImpl.java:125)
	at org.elixir_lang.beam.psi.impl.ModuleImpl.getNavigationElement(ModuleImpl.kt:105)
	at org.elixir_lang.psi.scope.module.Variants.projectNameElements(Variants.kt:126)
	at org.elixir_lang.psi.scope.module.Variants.access$projectNameElements(Variants.kt:23)
	at org.elixir_lang.psi.scope.module.Variants$Companion.lookupElementList(Variants.kt:153)
	at org.elixir_lang.reference.Module.getVariants(Module.kt:16)
	at org.elixir_lang.reference.Module.getVariants(Module.kt:13)
	at com.intellij.codeInsight.completion.CompletionData.completeReference(CompletionData.java:245)
	at com.intellij.codeInsight.completion.CompletionVariant.addReferenceCompletions(CompletionVariant.java:125)
	at com.intellij.codeInsight.completion.CompletionData.completeReference(CompletionData.java:66)
	at com.intellij.codeInsight.completion.LegacyCompletionContributor.lambda$completeReference$0(LegacyCompletionContributor.java:75)
	at com.intellij.codeInsight.completion.LegacyCompletionContributor.processReference(LegacyCompletionContributor.java:129)
	at com.intellij.codeInsight.completion.LegacyCompletionContributor.processReferences(LegacyCompletionContributor.java:112)
	at com.intellij.codeInsight.completion.LegacyCompletionContributor.completeReference(LegacyCompletionContributor.java:73)
	at com.intellij.codeInsight.completion.LegacyCompletionContributor.fillCompletionVariants(LegacyCompletionContributor.java:54)
	at com.intellij.codeInsight.completion.CompletionService.getVariantsFromContributors(CompletionService.java:77)
	at com.intellij.codeInsight.completion.CompletionResultSet.runRemainingContributors(CompletionResultSet.java:154)
	at com.intellij.codeInsight.completion.CompletionResultSet.runRemainingContributors(CompletionResultSet.java:146)
	at com.intellij.codeInsight.completion.CompletionResultSet.runRemainingContributors(CompletionResultSet.java:142)
	at com.intellij.codeInsight.template.impl.LiveTemplateCompletionContributor$1.addCompletions(LiveTemplateCompletionContributor.java:88)
	at com.intellij.codeInsight.completion.CompletionProvider.addCompletionVariants(CompletionProvider.java:32)
	at com.intellij.codeInsight.completion.CompletionContributor.fillCompletionVariants(CompletionContributor.java:155)
	at com.intellij.codeInsight.completion.CompletionService.getVariantsFromContributors(CompletionService.java:77)
	at com.intellij.codeInsight.completion.CompletionService.getVariantsFromContributors(CompletionService.java:60)
	at com.intellij.codeInsight.completion.CompletionService.performCompletion(CompletionService.java:133)
	at com.intellij.codeInsight.completion.BaseCompletionService.performCompletion(BaseCompletionService.java:41)
	at com.intellij.codeInsight.completion.CompletionProgressIndicator.lambda$calculateItems$12(CompletionProgressIndicator.java:862)
	at com.intellij.util.indexing.FileBasedIndex.lambda$ignoreDumbMode$0(FileBasedIndex.java:179)
	at com.intellij.openapi.util.RecursionManager$1.computePreventingRecursion(RecursionManager.java:111)
	at com.intellij.util.indexing.FileBasedIndexEx.ignoreDumbMode(FileBasedIndexEx.java:543)
	at com.intellij.util.indexing.FileBasedIndex.ignoreDumbMode(FileBasedIndex.java:178)
	at com.intellij.codeInsight.completion.CompletionProgressIndicator.calculateItems(CompletionProgressIndicator.java:858)
	at com.intellij.codeInsight.completion.CompletionProgressIndicator.runContributors(CompletionProgressIndicator.java:846)
	at com.intellij.codeInsight.completion.CodeCompletionHandlerBase.lambda$startContributorThread$6(CodeCompletionHandlerBase.java:353)
	at com.intellij.codeInsight.completion.AsyncCompletion.lambda$tryReadOrCancel$5(CompletionThreading.java:172)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1137)
	at com.intellij.codeInsight.completion.AsyncCompletion.tryReadOrCancel(CompletionThreading.java:170)
	at com.intellij.codeInsight.completion.CodeCompletionHandlerBase.lambda$startContributorThread$7(CodeCompletionHandlerBase.java:345)
	at com.intellij.codeInsight.completion.AsyncCompletion.lambda$startThread$0(CompletionThreading.java:95)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:178)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:658)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:610)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:65)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:165)
	at com.intellij.codeInsight.completion.AsyncCompletion.lambda$startThread$1(CompletionThreading.java:91)
	at com.intellij.util.RunnableCallable.call(RunnableCallable.java:20)
	at com.intellij.util.RunnableCallable.call(RunnableCallable.java:11)
	at com.intellij.openapi.application.impl.ApplicationImpl$1.call(ApplicationImpl.java:270)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:668)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:665)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:665)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

**Comment by angelikatyborska** [comment]
I keep running into the same issue (plugin version 11.9.1, RubyMine 2020.3.1 Build #RM-203.6682.166, built on December 29, 2020, macOS 10.15.7). I am also experiencing a very annoying issue that _I suspect_ is the outcome of this exception. On top of syntax highlighting breaking often like mentioned https://github.com/KronicDeth/intellij-elixir/issues/1889, sometimes I am unable to, among others, double-click on a word to select it or insert a dot. Pressing the dot key doesn't produce a character. Triple-click to select the whole line still works.

Here's a gif of how that looks like:

![2021-01-22 15 01 03](https://user-images.githubusercontent.com/7852553/105500859-c9443280-5cc3-11eb-9f5f-34d498ba4327.gif)

I need to restart Rubymine to be able to continue working.

**Comment by KronicDeth** [comment]
@angelikatyborska I also get the `.` not insertable problem.  Weirdly, I can type a `,` and then replace it with `.` using IDEAVim's r+, commands, so it's insertable in some cases, but not directly typable.

Can you try 
[intellij-elixir-11.9.2-pre+20210122060523.zip](https://github.com/KronicDeth/intellij-elixir/files/5856815/intellij-elixir-11.9.2-pre%2B20210122060523.zip) and see if it fixes the problem?

**Comment by angelikatyborska** [comment]
@KronicDeth I don't really have a reliable way to reproduce any of those problems, they just happen sometimes and I can't tell why. I installed this version and I will let you know _when_ either issue happens again, but that might take a while because at this moment, it's Friday evening for me, which might mean no more intensive Elixir coding until Monday morning 😄. In the last 5 minutes I tried to make it break, neither "No decompiled source function with name" nor "cannot type a dot" happened, but I did see a new error, `kotlin.KotlinNullPointerException`:

```
kotlin.KotlinNullPointerException
	at org.elixir_lang.psi.impl.call.CallImplKt.finalArguments(CallImpl.kt:120)
	at org.elixir_lang.psi.impl.call.CallImplKt.keywordArguments(CallImpl.kt:144)
	at org.elixir_lang.psi.impl.call.CallImplKt.keywordArgument(CallImpl.kt:135)
	at org.elixir_lang.psi.impl.call.CallImpl.hasDoBlockOrKeyword(CallImpl.kt:374)
	at org.elixir_lang.psi.impl.ElixirPsiImplUtil.hasDoBlockOrKeyword(ElixirPsiImplUtil.java:1136)
	at org.elixir_lang.psi.impl.ElixirUnmatchedMatchOperationImpl.hasDoBlockOrKeyword(ElixirUnmatchedMatchOperationImpl.java:78)
	at org.elixir_lang.annotator.Parameter.putParameterized(Parameter.java:86)
	at org.elixir_lang.annotator.Parameter.putParameterized(Parameter.java:136)
	at org.elixir_lang.annotator.Parameter.putParameterized(Parameter.java:42)
	at org.elixir_lang.reference.Callable$Companion.isParameter(Callable.kt:275)
	at org.elixir_lang.annotator.Callable.callHighlight(Callable.kt:187)
	at org.elixir_lang.annotator.Callable.callHighlight(Callable.kt:212)
	at org.elixir_lang.annotator.Callable.callHighlight(Callable.kt:226)
	at org.elixir_lang.annotator.Callable.highlight(Callable.kt:233)
	at org.elixir_lang.annotator.Callable.access$highlight(Callable.kt:28)
	at org.elixir_lang.annotator.Callable$annotate$1.visitPlainCall(Callable.kt:112)
	at org.elixir_lang.annotator.Callable$annotate$1.visitNonModuleAttributeCall(Callable.kt:87)
	at org.elixir_lang.annotator.Callable$annotate$1.visitCall(Callable.kt:47)
	at org.elixir_lang.annotator.Callable$annotate$1.visitElement(Callable.kt:78)
	at com.intellij.psi.impl.PsiElementBase.accept(PsiElementBase.java:273)
	at org.elixir_lang.psi.impl.ElixirUnmatchedUnqualifiedNoArgumentsCallImpl.accept(ElixirUnmatchedUnqualifiedNoArgumentsCallImpl.java:39)
	at org.elixir_lang.annotator.Callable.annotate(Callable.kt:39)
	at com.intellij.codeInsight.daemon.impl.DefaultHighlightVisitor.runAnnotators(DefaultHighlightVisitor.java:136)
	at com.intellij.codeInsight.daemon.impl.DefaultHighlightVisitor.visit(DefaultHighlightVisitor.java:116)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.runVisitors(GeneralHighlightingPass.java:336)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.lambda$collectHighlights$5(GeneralHighlightingPass.java:269)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.analyzeByVisitors(GeneralHighlightingPass.java:295)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.lambda$analyzeByVisitors$6(GeneralHighlightingPass.java:298)
	at com.intellij.codeInsight.daemon.impl.DefaultHighlightVisitor.analyze(DefaultHighlightVisitor.java:96)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.analyzeByVisitors(GeneralHighlightingPass.java:298)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.collectHighlights(GeneralHighlightingPass.java:266)
	at com.intellij.codeInsight.daemon.impl.GeneralHighlightingPass.collectInformationWithProgress(GeneralHighlightingPass.java:212)
	at com.intellij.codeInsight.daemon.impl.ProgressableTextEditorHighlightingPass.doCollectInformation(ProgressableTextEditorHighlightingPass.java:84)
	at com.intellij.codeHighlighting.TextEditorHighlightingPass.collectInformation(TextEditorHighlightingPass.java:56)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$1(PassExecutorService.java:400)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1137)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$doRun$2(PassExecutorService.java:393)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:658)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:610)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:65)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.doRun(PassExecutorService.java:392)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.lambda$run$0(PassExecutorService.java:368)
	at com.intellij.openapi.application.impl.ReadMostlyRWLock.executeByImpatientReader(ReadMostlyRWLock.java:172)
	at com.intellij.openapi.application.impl.ApplicationImpl.executeByImpatientReader(ApplicationImpl.java:183)
	at com.intellij.codeInsight.daemon.impl.PassExecutorService$ScheduledPass.run(PassExecutorService.java:366)
	at com.intellij.concurrency.JobLauncherImpl$VoidForkJoinTask$1.exec(JobLauncherImpl.java:188)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
```

**Comment by KronicDeth** [comment]
Fixed by #1887

**Comment by angelikatyborska** [comment]
I have been using the 11.9.2 release the whole day today it and I am not experiencing problems anymore 🎉
